compiler/rustc_span/src/def_id.rs

```diff
@@ -381,6 +381,12 @@ impl LocalDefId {
     }
 }
 
+impl From<DefId> for LocalDefId {
```

Comment:
Can this impl be avoided? `expect_local()` explicitly states that it may panic. This is not consistent with how `From` is expected not to fail.